Refactor service extensions and set button text based on extension state.

kenzieschmoll opened the pull request:
Service extension data is now stored in ServiceExtensions.java. Button text now reflects extension state.

Performance overlay example:
When the extension is enabled, the tooltip will now read "Hide Performance Overlay".
When the extension is disabled, the tooltip will now read "Show Performance Overlay".

This applies to all the service extension buttons, including those with visible text in the inspector dropdown menu.

jacob314 commented on src/io/flutter/view/FlutterView.java:
```diff
@@ -632,16 +632,18 @@ public void perform(AnActionEvent event) {
     super(app, "Toggle Select Widget Mode", "Toggle Select Widget Mode", AllIcons.General.LocateHover);
-
     setExtensionCommand("ext.flutter.inspector.show");
+    setEnabledText("Disable Select Widget Mode");
```
Extend the constructor of FlutterViewToggleableAction to take a couple more parameters. That way it will be harder to forget to add these.

kenzieschmoll replied:
Done.

jacob314 commented on src/io/flutter/view/FlutterView.java:
```diff
@@ -631,15 +631,27 @@ public void perform(AnActionEvent event) {
+      app,
+      "Show Repaint Rainbow",
+      "Hide Repaint Rainbow",
+      "Show Repaint Rainbow",
```
nit: make a constructor with two less parameters that lets you avoid repeating yourself
when the names match.

kenzieschmoll replied:
Refactored.

jacob314 commented on src/io/flutter/view/FlutterView.java:
```diff
@@ -631,15 +631,27 @@ public void perform(AnActionEvent event) {
-    super(app, "Toggle Select Widget Mode", "Toggle Select Widget Mode", AllIcons.General.LocateHover);
+    super(
+      app,
+      "Toggle Select Widget Mode",
```
why do these names diverge?

kenzieschmoll replied:
Refactored.

jacob314 commented on src/io/flutter/view/FlutterView.java:
```diff
@@ -704,7 +716,13 @@ public void update(@NotNull AnActionEvent e) {
+      app,
+      "Hide Debug Mode Banner",
+      "Show Debug Mode Banner",
+      "Hide Debug Mode Banner",
```
why are these both hide?

kenzieschmoll replied:
Refactored.

jacob314 commented on src/io/flutter/view/FlutterViewToggleableAction.java:
```diff
@@ -21,20 +21,39 @@
     this.extensionCommand = extensionCommand;
   }
 
+  protected void setEnabledText(String enabledText) {
```
remove setEnabledText and setDisabledText

jacob314 commented on src/io/flutter/view/TimeDilationAction.java:
```diff
@@ -16,7 +16,13 @@
-    super(app, "Enable Slow Animations", null, showIcon ? AllIcons.Vcs.History : null);
+    super(
+      app,
+      "Enable Slow Animations",
```
similarly would be nice to avoid writing Enable Slow Animations twice

kenzieschmoll commented:
Refactored all service extensions. Extensions are now stored in ServiceExtensions.java as ToggleableServiceExtensionDescription objects. Adjusted FlutterViewToggleableAction accordingly.

jacob314 commented on src/io/flutter/view/InspectorPerfTab.java:
```diff
@@ -76,8 +74,18 @@
+        ServiceExtensions.performanceOverlay.getDisabledText(),
+        "");
+    showRepaintRainbow =
+      new BoolServiceExtensionCheckbox(
```
make
BoolServiceExtensionCheckbox just take the whole extension description as an argument.
That way it can be later cleaned up to use more of the metadata provided and it will simplify this code a bit.
